crypto/x509/x_attrib.c

```diff
@@ -60,38 +60,28 @@ X509_ATTRIBUTE *X509_ATTRIBUTE_create(int nid, int atrtype, void *value)
 
 static int print_hex(BIO *out, unsigned char *buf, int len)
 {
-    int i;
+    int result = -1;
+    char *hexbuf;
+    int hexlen = len * 2 + 1;
 
-    for (i = 0; i < len; i++) {
-        if (BIO_printf(out, "%02X ", buf[i]) <= 0) {
-            return 0;
-        }
-    }
-    return 1;
-}
+    hexbuf = OPENSSL_malloc(hexlen);
+    if (hexbuf == NULL)
+        return 0;
+    result = OPENSSL_buf2hexstr_ex(hexbuf, hexlen, NULL, buf, len, ':');
+    if (result != 1)
+        goto err;
+    if ((result = BIO_puts(out, hexbuf)) <= 0)
+        goto err;
 
-static int asn1_integer_print_bio(BIO *bio, const ASN1_INTEGER *num)
-{
-    BIGNUM *num_bn;
-    int result = 0;
-    char *hex;
-
-    num_bn = ASN1_INTEGER_to_BN(num, NULL);
-    if (num_bn == NULL)
-        return -1;
-    if ((hex = BN_bn2hex(num_bn)) != NULL) {
-        result = BIO_write(bio, "0x", 2) > 0;
-        result = result && BIO_write(bio, hex, strlen(hex)) > 0;
-        OPENSSL_free(hex);
-    } else {
-        return -1;
-    }
-    BN_free(num_bn);
+    OPENSSL_free(hexbuf);
+    return 1;
 
-    return result;
+    err:
+        OPENSSL_free(hexbuf);
+        return 0;
```

Comment:
Bad indentation - `err:` should be indented just by one space and the two lines after the label by 4 spaces.